src/main/java/storagesystem/services/JSONHandler.java

```diff
@@ -0,0 +1,278 @@
+        Gson gson = gsonBuilder.setPrettyPrinting().create();
+        JsonArray jsonContent = gson.fromJson(new FileReader(fileName), JsonArray.class);
+        Writer writer = new FileWriter(fileName);
+        JsonObject jsonObject;
+        if (jsonContent == null) {
+            jsonContent = new JsonArray();
+        }
+        jsonObject = (JsonObject) gson.toJsonTree(objectToAdd);
+        jsonContent.add(jsonObject);
+        gson.toJson(jsonContent, writer);
+
+        writer.flush();
+        writer.close();
+    }
+
+    /**
+     * addListToJson() adds a list of choice to a json file of choice without erasing the contents that are already in the json file.
+     *
+     * @param listToAdd
+     * @throws IOException
+     */
+
+    public static void addListToJson(List listToAdd) throws IOException {
+        try {
+            if (listToAdd.get(0) instanceof Item) {
+                addList(listToAdd, itemDB);
+            } else if (listToAdd.get(0) instanceof Location) {
+                addList(listToAdd, locationDB);
+            } else if (listToAdd.get(0) instanceof Organisation) {
+                addList(listToAdd, organisationDB);
+            } else if (listToAdd.get(0) instanceof Reservation) {
+                addList(listToAdd, reservationDB);
+            } else if (listToAdd.get(0) instanceof Team) {
+                addList(listToAdd, teamDB);
+            } else if (listToAdd.get(0) instanceof User) {
+                addList(listToAdd, userDB);
+            }
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("List is empty!");
+        }
+    }
+
+    /**
+     * @param listToAdd
+     * @param database
+     * @throws IOException
+     */
+    private static void addList(List listToAdd, String database) throws IOException {
+        for (Object object : listToAdd) {
+            addToJson(object, database);
+        }
+    }
+
+    /**
+     * clearJson() clears a json file completely.
+     *
+     * @param fileName
+     * @throws IOException
+     */
+
+    public static void clearJson(String fileName) throws IOException {
+        Writer writer = new FileWriter(fileName);
+        writer.flush();
+        writer.close();
+    }
+
+    /**
+     * clearAllJsonFiles() clears all json files declared completely.
+     *
+     * @throws IOException
+     */
+
+    public static void clearAllJsonFiles() throws IOException {
+        clearJson(itemDB);
+        clearJson(locationDB);
+        clearJson(organisationDB);
+        clearJson(reservationDB);
+        clearJson(teamDB);
+        clearJson(userDB);
+        System.out.println("CLEARED ALL JSON FILES");
+    }
+
+
+    /**
+     * Uses private methods to get a list.
+     *
+     * @param typeOfList
+     * @return a list with objects from a json file corresponding to typeOfList
+     * @throws IOException
+     */
+
+    public static List getListFromJson(Type typeOfList) throws IOException {
```

Comment:
Is it possible to set type for List? Unsure how this works

Reply:
For item we use getListFromJson(Item.class) and get a List containing Items.